dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandler.kt

```diff
@@ -47,45 +42,38 @@ internal class LogcatLogHandler(
         }
     }
 
-    private fun resolveTag(): String {
-        return if (Datadog.isDebug) {
-            return resolveTagFromCallerClassName()
-        } else {
-            serviceName
-        }
-    }
-
-    private fun resolveTagFromCallerClassName(): String {
-        val className: String = tryAndSearchClassName() ?: return serviceName
-        var tag = stripAnonymousPart(className)
-        tag = tag.substring(tag.lastIndexOf('.') + 1)
+    // endregion
 
-        return sanitizeTag(tag)
-    }
+    // region Internal
 
-    private fun tryAndSearchClassName(): String? {
-        val stackTrace = Throwable().stackTrace
-        // it might happen that when called from Java code the stack index to be ++
-        // due to the Java - Kotlin bridge method so we need an extra check here.
-        for (i in callerNameStackIndex until stackTrace.size) {
-            val className = stackTrace[i].className
-            if (className != LOGGER_CLASS_NAME)
-                return className
+    private fun resolveTag(stackTraceElement: StackTraceElement?): String {
+        return if (stackTraceElement == null) {
+            serviceName
+        } else {
+            stackTraceElement.className
+                .replace(ANONYMOUS_CLASS, "")
+                .substringAfterLast('.')
         }
-        return null
     }
 
-    private fun sanitizeTag(tag: String): String {
-        return if (tag.length < MAX_TAG_LENGTH || Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            tag
+    private fun resolveSuffix(stackTraceElement: StackTraceElement?): String {
+        return if (stackTraceElement == null) {
+            ""
         } else {
-            tag.substring(0, MAX_TAG_LENGTH)
+            "\t| at .${stackTraceElement.methodName}" +
+                "(${stackTraceElement.fileName}:${stackTraceElement.lineNumber})"
         }
     }
 
-    private fun stripAnonymousPart(className: String): String {
-        val matcher = ANONYMOUS_CLASS.matcher(className)
-        return matcher.replaceAll("")
+    private fun getCallerStackElement(): StackTraceElement? {
+        return if (Datadog.isDebug) {
+            val stackTrace = Throwable().stackTrace
+            stackTrace.firstOrNull {
+                it.className !in ignoredClassNames
```

Comment:
I think this is better even though it is less optimized as it will start the iteration from 0 but I can accept this and remove that magic "nestedDepth" .